contracts/periphery/src/query.rs

```diff
@@ -0,0 +1,160 @@
+        mint_amount,
+        mint_spend_amount,
+        swap_result_amount: coin(sim_res.total_input.u128(), input_asset),
+    })
+}
+
+pub fn simulate_burn_exact_amount_in(
+    deps: Deps,
+    _env: Env,
+    core_addr: String,
+    index_amount: Uint128,
+    output_asset: String,
+    swap_info: Vec<SwapInfo>,
+) -> Result<SimulateBurnExactAmountInResponse, ContractError> {
+    // query to core contract
+    let core = IbcCore(deps.api.addr_validate(&core_addr)?);
+    let core_fee = core.get_fee(&deps.querier, None)?;
+    let core_portfolio = core.get_portfolio(&deps.querier, None)?;
+
+    let pool_ids = extract_pool_ids(swap_info.clone());
+    let pools = query_pools(&deps, pool_ids)?;
+
+    // apply burn fee before simulating
+    let burn_amount = index_amount * deduct_fee(core_fee.burn_fee)?;
+
+    let conv = make_unit_converter(burn_amount);
+    let mut burn_redeem_amount: Vec<_> =
+        core_portfolio.units.clone().into_iter().map(conv).collect();
+    burn_redeem_amount.sort_by(|a, b| a.denom.cmp(&b.denom));
+
+    let sim = Simulator::new(&deps, &pools, &swap_info, &core_portfolio.units);
+    let sim_res = sim.estimate_output_for_index(burn_amount, &output_asset)?;
+
+    Ok(SimulateBurnExactAmountInResponse {
+        burn_amount,
+        burn_redeem_amount,
+        swap_result_amount: coin(sim_res.total_output.u128(), &output_asset),
+    })
+}
+
+pub fn simulate_burn_exact_amount_out(
+    deps: Deps,
+    _env: Env,
+    core_addr: String,
+    output_asset: Coin,
+    swap_info: Vec<SwapInfo>,
+) -> Result<SimulateBurnExactAmountOutResponse, ContractError> {
+    // query to core contract
+    let core = IbcCore(deps.api.addr_validate(&core_addr)?);
+    let core_fee = core.get_fee(&deps.querier, None)?;
+    let core_portfolio = core.get_portfolio(&deps.querier, None)?;
+
+    let pool_ids = extract_pool_ids(swap_info.clone());
+    let pools = query_pools(&deps, pool_ids)?;
+
+    let sim = Simulator::new(&deps, &pools, &swap_info, &core_portfolio.units);
+    let sim_res = sim
+        .estimate_index_for_output(output_asset.clone(), None, None, None)?
+        .min;
+
+    // apply burn fee after simulation
+    let burn_amount = sim_res.est_min_token_in * expand_fee(core_fee.burn_fee)?;
+
+    let conv = make_unit_converter(burn_amount);
+    let mut burn_redeem_amount: Vec<_> = core_portfolio.units.into_iter().map(conv).collect();
+    burn_redeem_amount.sort_by(|a, b| a.denom.cmp(&b.denom));
+
+    Ok(SimulateBurnExactAmountOutResponse {
+        burn_amount,
+        burn_redeem_amount,
+        swap_result_amount: coin(sim_res.max_token_out.u128(), output_asset.denom),
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use cosmwasm_std::Decimal;
+
+    #[test]
+    fn test_sad() {
+        let invert = Decimal::one()
+            .checked_div(Decimal::from_ratio(5u64, 1000u64))
+            .unwrap();
+        assert_eq!(invert, Decimal::from_ratio(200u64, 1u64));
+    }
+}
```

Comment:
It seems unnecessary

Reply:
just checking :)

Reply:
@byeongsu-hong If you think this test is necessary, please change name from `sad` to other thing.

Reply:
https://github.com/many-things/ibcx-contracts/pull/68/commits/f7fd7d5db898d7c508a7523baa19d5e927272729